test/devise/async/model_test.rb

```diff
@@ -3,59 +3,79 @@
 module Devise
   module Async
     describe "Model" do
+
+      # As per the https://github.com/plataformatec/devise/blob/master/CHANGELOG.md#410,
+      # within transaction mail will be enqued & after commit send_devise_pending_notifications will be invoked
+
+      # 1. Within transaction i.e object creation, mails will only be accummulated 
       it "accumulates notifications to be sent after commit on Model creation" do
+        admin = create_admin
         Admin.transaction do
-          admin = create_admin
           mailers = admin.send(:devise_pending_notifications) # [:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]
-          mailers.size.must_equal 1
+          mailers.size.must_equal 0
+          Worker.expects(:enqueue).never # after_commit will not fire without save
+        end  
+        admin[:username] = "changed_username"
+        admin.send_confirmation_instructions
 
-          mailer = mailers.first
-          mailer.size.must_equal 2
-          mailer.first.must_equal :confirmation_instructions
-          mailer.last.must_be_instance_of Array
-        end
+        mailers = admin.send(:devise_pending_notifications) # [[:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]]
+        mailer = mailers.first              # [:confirmation_instructions, ["RUQUib67wLcCiEyZMwfx", {}]]
+        notification_type = mailer.first    # :confirmation_instructions
+        args = mailer.last                  # ["RUQUib67wLcCiEyZMwfx", {}]
+
+        mailers.size.must_equal 1
+        mailer.size.must_equal 2
+        notification_type.must_equal :confirmation_instructions
+        args.must_be_instance_of Array      
+        Worker.expects(:enqueue).never # after_commit will not fire without save
       end
 
+      # 2. On explicit save, after commit will be invoked hence notifications will be queued
       it "immediately sends notifications when the model has not changed" do
-        admin = create_admin
-        Worker.expects(:enqueue).with(:confirmation_instructions, "Admin", admin.id.to_s, instance_of(String), {})
-        admin.send_confirmation_instructions
+        Admin.transaction do
+          @admin2 = create_admin
```

Comment:
Why does this need to be an instance variable? I guess `admin` would be a more suitable name here as there is only one admin been created.